assets/js/components/EntityHeader.js

```diff
@@ -0,0 +1,158 @@
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { useWindowScroll } from 'react-use';
+import throttle from 'lodash/throttle';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	useContext,
+	useCallback,
+	useEffect,
+	useRef,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import ViewContextContext from './Root/ViewContextContext';
+import { VIEW_CONTEXT_PAGE_DASHBOARD } from '../googlesitekit/constants';
+import Button from './Button';
+import { CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import BackspaceIcon from '../../svg/keyboard-backspace.svg';
+import { CORE_LOCATION } from '../googlesitekit/datastore/location/constants';
+import Link from './Link';
+const { useSelect, useDispatch } = Data;
+
+const EntityHeader = () => {
+	const viewContext = useContext( ViewContextContext );
+	const currentEntityTitle = useSelect( ( select ) =>
+		select( CORE_SITE ).getCurrentEntityTitle()
+	);
+	const entityURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getCurrentEntityURL()
+	);
+
+	const headerDetailsRef = useRef();
+	const [ url, setURL ] = useState( entityURL );
+
+	useEffect( () => {
+		const shortenURL = () => {
+			if ( ! headerDetailsRef.current ) {
+				return;
+			}
+
+			// Remove 40 px for margins + SVG at the end of the URL link
+			const availableWidth = headerDetailsRef.current.clientWidth - 40;
+
+			const urlFontSize = global
+				.getComputedStyle( headerDetailsRef.current.lastChild, null )
+				.getPropertyValue( 'font-size' );
+			const fontSize = parseFloat( urlFontSize );
+
+			// 2 is appox. the minimum character constant for sans-serif fonts:
+			// https://pearsonified.com/characters-per-line/
+			const maxChars = ( availableWidth * 2 ) / fontSize;
+
+			const shortenedURL = new URL( entityURL );
+
+			if ( maxChars < entityURL.length ) {
+				const extraChars = entityURL.length - maxChars + 4; // length of "/..." = 4
+				const origin = shortenedURL.origin;
+				const restOfURL = shortenedURL.toString().replace( origin, '' );
+				const newRestOfURL = '/...' + restOfURL.substr( extraChars );
+				setURL( origin + newRestOfURL );
+			} else {
+				setURL( entityURL );
+			}
+		};
+
+		// Use throttled version only on window resize
```

Comment:
Single line comments should start with a capital letter and end with a full stop:

```suggestion
		// Use throttled version only on window resize.
```